snooty/parser.py

```diff
@@ -292,7 +292,7 @@ def dispatch_visit(self, node: docutils.nodes.Node) -> None:
                 self.state.append(role)
                 return
 
-            elif role_name == "icon":
+            elif role_name.startswith("icon"):
```

Comment:
A part of me feels like this could become too permissive, but not a large concern and improbable that we'd add an `icon` prefix'd directive without wanting to add it to validation - so I'm fine with this and don't think we need to futureproof too much.